apphub/NLP/named_entity_recognition/bert_torch.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import tempfile
+from typing import Callable, Iterable, Union
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as fn
+from transformers import BertConfig, BertModel, BertTokenizer
+
+import fastestimator as fe
+from fastestimator.dataset.data import german_ner
+from fastestimator.op.numpyop.numpyop import NumpyOp
+from fastestimator.op.numpyop.univariate import PadSequence, Tokenize, WordtoId
+from fastestimator.op.tensorop import TensorOp
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.trace.io import BestModelSaver
+from fastestimator.trace.metric import Accuracy
+
+
+def char2idx(data):
+    """char2idx function creates look-up table for the corresponding ids for the labels"""
+    tag2idx = {t: i for i, t in enumerate(data)}
+    return tag2idx
+
+
+class AttentionMask(NumpyOp):
+    """Custom NumpyOp that constructs attention masks for input sequences"""
+    def forward(self, data, state):
+        masks = [float(i > 0) for i in data]
+        return np.array(masks)
+
+
+class ReshapeOp(TensorOp):
```

Comment:
is there some reason why reshape should not be a generic (FE) op taking shape as an arg?